test/file_io.jl

```diff
@@ -170,6 +170,35 @@ end
             JLSO.save("breakfast.jlso", "food" => "☕️🥓🍳", "time" => Time(9, 0)),
         )
     end
+    @testset "Size Limits" begin
+        # Tests that ensure we can work with objects larger than the BSON max
+        # doc/string/array size. Limited to size typemax(Int32).
+        # Instead of having a file size limited to 2.14 GB we should now support petabyte
+        # sized objects.
+        mktempdir() do d
+            @testset "Large Objects" begin
+                obj = zeros(UInt8, typemax(Int32) + 1)
+                JLSO.save(
+                    joinpath(d, "large-object.jlso"),
+                    :X => obj;
+                    compression=:none,
+                )
+                loaded = JLSO.load(joinpath(d, "large-object.jlso"))
+                @test loaded == Dict(:X => obj)
+            end
+            @testset "Large Docs" begin
```

Comment:
I don't understand why this is called `large docs`

Reply:
Rather than having a single large vector (usually where the error hit) we want to test that the same issue occurs for large BSON docs. I suppose in BSON.jl the doc is called `BSONDict` so we could change it to that.

Reply:
or just large total size?

Reply:
Yeah, cause the max buffer length cannot exceed `typemax(Int32)` https://github.com/JuliaIO/BSON.jl/blob/master/src/write.jl#L32

Reply:
```suggestion
            @testset "Large Total Size" begin
```